dialect/sql/sqlgraph/graph.go

```diff
@@ -271,6 +271,7 @@ func HasNeighborsWith(q *sql.Selector, s *Step, pred func(*sql.Selector)) {
 			Join(to).
 			On(edge.C(pk1), to.C(s.To.Column))
 		matches := builder.Select().From(to)
+		matches.WithContext(q.Context())
```

Comment:
@a8m this is needed because `matches` is what actually gets passed to `pred(matches)` right below which are the `Has` and `HasWith` functions that _need_ the Selector to have a context to extract the "schemaconfig" and apply it to edges.